### ISSUE TITLE
Feature request: "delete all containers and reset to default containers" option

### ISSUE THREAD

**Issue by Valtarien** [mozilla/multi-account-containers#1977]
### TL:DR
**In summary, I request the implementation of a "delete all containers and reset to defaults" feature in the extension preferences, and option upon re-installing the extension. This should be a fairly straightforward solution to resolve not only problems resulting from the "[Temporary Containers](https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/)" extension, but any other problematic extension that exists or gets developed in the future.

Also, is there any way to manually delete and reset all containers to a default new installation configuration?**

- **YES:** "Firefox will: Never remember history" in the Firefox Preferences/Options under "Privacy & Security > History" is NOT selected
- **YES:** You are NOT using Firefox in a Private Window
- **YES:** You can see a grayed out but ticked Checkbox with the description "Enable Container Tabs" in the Firefox Preferences/Options under "Tabs"

### Add-ons
- Multi-Account Containers Version: 7.3.0
- Operating System + Version: Pop!_OS 20.10 (Ubuntu-based Linux distribution) and Windows 10 (updated fully to 2021-02-26).
- Firefox Version: 85.0.1 (64-bit)
- Other installed Add-ons + Version + Enabled/Disabled-Status:
Name 	Type 	Version 	Enabled 	ID
Amazon Smile Redirect	extension	2.0.11	true	{aac801ed-5288-41e0-a937-7aa70207ab13}
Amazon.com	extension	1.3	true	amazondotcom@search.mozilla.org
Auto Tab Discard	extension	0.4.1.1	true	{c2c003ee-bd69-42a2-b0e9-6f34222cb046}
Auto-Sort Bookmarks	extension	3.4.1	true	sortbookmarks@bouanto
Bing	extension	1.3	true	bing@search.mozilla.org
ClearURLs	extension	1.20.0	true	{74145f27-f039-47ce-a470-a662b129930a}
Cookie AutoDelete	extension	3.6.0	true	CookieAutoDelete@kennydo.com
Copy PlainText	extension	1.11	true	copyplaintext@eros.man
Copy ShortURL	extension	5.1.1	true	jid0-ODIKJS9b4IT3H1NYlPKr0NDtLuE@jetpack
CSS Exfil Protection	extension	1.1.0	true	{7fc8ef53-24ec-4205-87a4-1e745953bb0d}
Decentraleyes	extension	2.0.15	true	jid1-BoFifL9Vbdl2zQ@jetpack
Disable JavaScript	extension	2.3.1	true	{41f9e51d-35e4-4b29-af66-422ff81c8b41}
Don't Fuck With Paste	extension	2.7	true	DontFuckWithPaste@raim.ist
DownThemAll!	extension	4.2.6	true	{DDC359D1-844A-42a7-9AA1-88A850A938A8}
DuckDuckGo	extension	1.1	true	ddg@search.mozilla.org
eBay	extension	1.3	true	ebay@search.mozilla.org
Firefox Multi-Account Containers	extension	7.3.0	true	@testpilot-containers
Firefox Relay	extension	1.5.1	true	private-relay@firefox.com
Google	extension	1.1	true	google@search.mozilla.org
Grammar and Spell Checker — LanguageTool	extension	3.3.4	true	languagetool-webextension@languagetool.org
HTTPS Everywhere	extension	2021.1.27	true	https-everywhere@eff.org
IBA Opt-out (by Google)	extension	2.0	true	optout@google.com
Joplin Web Clipper	extension	1.6.1	true	{8419486a-54e9-11e8-9401-ac9e17909436}
Link Gopher	extension	2.0.1	true	linkgopher@oooninja.com
Lustre	extension	1.7.6	true	{8f9bcce0-80fe-4076-b8b1-988a81152068}
MarkDownload - Markdown Web Clipper	extension	3.0.0	true	{1c5e4c6f-5530-49a3-b216-31ce7d744db0}
Media Bias/Fact Check Extension	extension	3.1.2	true	media-bias-fact-check@drmikecrowe-mbfcext
Minimal Consent	extension	1.0.9	true	{94c24b3e-bf92-4ace-9279-e7fbf4dbf186}
Official Media Bias Fact Check Icon	extension	1.7.1	true	{6ba00d80-d8f3-4e1a-bc70-e1b6c64ae4a0}
Offline QR Code Generator	extension	1.7	true	offline-qr-code@rugk.github.io
OneNote Web Clipper	extension	3.8.0	true	Clipper@OneNote.com
OneTab	extension	1.50	true	extension@one-tab.com
Polisis	extension	2.0	true	{54da8dcf-a015-4c80-bbb4-db7f226ac7bd}
Privacy Badger	extension	2021.2.2	true	jid1-MnnxcxisBPnSXQ@jetpack
Privacy Possum	extension	2019.7.18	true	woop-NoopscooPsnSXQ@jetpack
RegretsReporter	extension	1.0.1	true	regrets-reporter@mozillafoundation.org
SingleFile	extension	1.18.69	true	{531906d3-e22f-4a6c-a102-8057b88a1a63}
Stylus	extension	1.5.17	true	{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}
Swift Selection Search	extension	3.47.3	true	jid1-KdTtiCj6wxVAFA@jetpack
Terms of Service; Didn’t Read	extension	4.0.6	true	jid0-3GUEt1r69sQNSrca5p8kx9Ezc3U@jetpack
TinEye Reverse Image Search	extension	1.4.1	true	tineye@ideeinc.com
Tranquility Reader	extension	3.0.22	true	tranquility@ushnisha.com
uBlock Origin	extension	1.33.2	true	uBlock0@raymondhill.net
Unpaywall	extension	3.94	true	{f209234a-76f0-4735-9920-eb62507a54cd}
VirusTotal Scan	extension	0.6.3	true	{06def8b7-bcc3-4f1e-add7-7d8222ca0efc}
Web Clipper	extension	1.29.0	true	{3fbb1f97-0acf-49a0-8348-36e91bef22ea}
Wikipedia (en)	extension	1.1	true	wikipedia@search.mozilla.org
Youtube Audio	extension	0.0.2.5	true	{580efa7d-66f9-474d-857a-8e2afc6b1181}
Zoom Page WE	extension	17.13	true	zoompage-we@DW-dev
English (CA) Language Pack	locale	85.0buildid20210204182252	true	langpack-en-CA@firefox.mozilla.org
English (GB) Language Pack	locale	85.0buildid20210204182252	true	langpack-en-GB@firefox.mozilla.org

### Actual behavior
After uninstalling the "[Temporary Containers](https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/)" (TC) extension, there were hundreds of temporary containers ("tmp###") left behind. After uninstalling "Multi-Account Containers" (MAC) and all dependent extensions that relied on containers, deleting my cache and profile, recreating, re-syncing Firefox (FF), and reinstalling MAC, I still have all those unwanted temporary containers.

### Expected behavior
Uninstalling and reinstalling MAC should provide an option to reset containers to default initial offerings. Additionally, it would be ideal to allow for a "delete all containers and reset to defaults" option in the extension preferences.

### Steps to reproduce
1. Install MAC.
2. Install "[Temporary Containers](https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/)" (TC) extension and start browsing.
3. After visiting several sites, allowing for the creation of temporary containers, uninstall both TC and MAC.
4. Delete entire profile and restart FF.
5. Re-sync your profile.
6. Re-install MAC and notice that all temporary containers remain, with no option but to delete them one-by-one, involving multiple clicks for each deletion.

**Comment by Valtarien** [comment]
After further investigation, I see that this problem is a recurring one for many people, at least in the following posted issues:

- https://github.com/stoically/temporary-containers/issues/371
- https://github.com/stoically/temporary-containers/issues/1675
- https://github.com/mozilla/multi-account-containers/issues/1912
- https://github.com/mozilla/multi-account-containers/issues/1885
- https://github.com/mozilla/multi-account-containers/issues/1839
- https://github.com/mozilla/multi-account-containers/issues/1799
- https://github.com/mozilla/multi-account-containers/issues/1737
- https://github.com/mozilla/multi-account-containers/issues/1732
- https://github.com/mozilla/multi-account-containers/issues/1661

With so many posts about this, and the unresolved nature of it, it is no wonder I've not gotten a response here. Still, I would appreciate it if someone who finds a solution to this problem would notify me here in a PM. I have managed to get rid of my own tmp* containers (over a thousand!) by utilizing various proposed solutions, disabling sync, uninstalling all extensions affecting containers, re-syncing Firefox, re-installing, deleting recurring synced tmp* containers, etc. It is time-consuming and frustrating, to say the least. I will surely never again use "Temporary Containers" after this experience, at least until the developers find a way to prevent the temp* containers from syncing and cluttering up the containers list.

**Comment by onebod** [comment]
Same problem for years now.
Recently found an option to delete containers with less hassle:
about:preferences#containers

Sync just puts them back though.
The ability to wipe the entire thing and start over is imperative for anyone who has used this and Temporary Containers extension.

Another confounding annoyance is using with Cookie Auto-Delete which does not sync also means every site I want to allow cookies must be set manually in every container until I export CA-D to the new machine.
In the meantime the terrible and inconsistent way M-AC works means I'm constantly ignoring or messing around with containers and CA-D options.

The cognitive overhead and gnawing background concern mean I am likely to remove M-AC which is a shame because the basic idea is valuable because it is important and the basic intention is entirely overwhelmed by the ridiculous function of syncing containers that were supposed to be temporary, can easily programmatically be identified as such and have been repeatedly been deleted manually.

**Comment by AbdulR3hman** [comment]
@onebod this was so helpful thank you; honestly had over 1000 container; was impossible to delete through normal mechanism; this made it a tad better helpful! Just used a mouse clicker 😆

**Comment by go2null** [comment]
Found this add-on which allows deleting multiple containers by regex: https://addons.mozilla.org/en-US/firefox/addon/containers-helper
